### PR TITLE
FLEX-5358: Send for every relay switch a notification

### PR DESCRIPTION
**Pull request by joostknapen**

### CODE REVIEW

File: integration-tests/cucumber-tests-platform-publiclighting/src/test/java/org/opensmartgridplatform/cucumber/platform/publiclighting/mocks/oslpdevice/MockOslpChannelHandler.java [hunk 1]
```diff
@@ -262,6 +220,51 @@ public void channelRead0(final ChannelHandlerContext ctx, final OslpEnvelope mes
...
+
+
+    private OslpEnvelope buildNotification(final OslpEnvelope message, final String deviceUid,
+            final DeviceState deviceState, final Oslp.Event event, final ByteString index) {
+        this.devicesContext.getDeviceState(deviceUid).incrementSequenceNumber();
+
+        final Oslp.EventNotification.Builder eventNotificationBuilder = Oslp.EventNotification.newBuilder()
+                .setEvent(event)
+                .setDescription(event.name())
+                //.setTimestamp(PlatformDefaults.TIMESTAMP)
```
Comment (joostknapen) [inline integration-tests/cucumber-tests-platform-publiclighting/src/test/java/org/opensmartgridplatform/cucumber/platform/publiclighting/mocks/oslpdevice/MockOslpChannelHandler.java:251]: Default timestamp ="now", but cannot be parsed by OSGP. What is the correct timestamp format? (new Date() doesn't work) The former cucumber scenario also didn't add the timestamp to the notification.

**Comment by joostknapen** [comment]
retest this please

File: integration-tests/cucumber-tests-platform-publiclighting/src/test/java/org/opensmartgridplatform/cucumber/platform/publiclighting/mocks/oslpdevice/MockOslpChannelHandler.java [hunk 2]
```diff
@@ -142,6 +78,23 @@ public MockOslpChannelHandler(final String oslpSignature, final String oslpSigna
         this.receivedResponses = receivedResponses;
     }
 
+    private static byte[] convertIntegerToByteArray(final Integer value) {
+        // See: platform.service.SequenceNumberUtils
```
[the remark this answers is not in the document]
Comment (rlemmers) [reply integration-tests/cucumber-tests-platform-publiclighting/src/test/java/org/opensmartgridplatform/cucumber/platform/publiclighting/mocks/oslpdevice/MockOslpChannelHandler.java:86]: We'll leave it in, to limit the coupling.

**Comment by rlemmers** [comment]
In my latest changes, I also made some changes to improve the SetLight.feature:

- Multiple light values were specified for the same relay.
- We normally use relay 1 for tariff, so in the tests it's more appropriate to start with relay 2.
- The test cases for multiple light values could be reduced from 4 to 3 examples.

**Pull request merged**